### PR TITLE
Add `mermaid` based flow charts for the crucial fast-sync processes

### LINKED ISSUE #3377: Add flow charts to new flows introduced in "new" fast sync implementation
Description: Use the mermaid format and then the aquamarine crate to create proper docs
Points: 5

### PR DESCRIPTION
**Pull request by rafal-ch**
This PR `mermaid` based flow charts for the crucial fast-sync processes.

Please use the tool available at https://mermaid.live/ to preview the diagrams.

Docs can be built with `cargo doc --document-private-items`

Closes #3377

### CODE REVIEW

**Comment by rafal-ch** [comment]
Added "do-not-merge" - I have an idea how to make the component diagram more legible.

File: node/src/components/sync_leaper.rs [hunk 1]
```diff
@@ -268,6 +268,29 @@ impl SyncLeaper {
...
+    ///
+    ///     Start --> A{have at least<br>one peer?}
+    ///     A -->|Yes| B{is other sync<br>leap in progress?}
+    ///     A -->|No| End
+    ///     B -->|Yes| C{do sync leap<br>identifiers match?}
+    ///     C -->|No| End
+    ///     C -->|Yes| D[fetch SyncLeap from potentially<br>newly learned peers]
+    ///     B -->|No| G[fetch SyncLeap<br>from all peers]
+    ///     G --> E
+    ///     D -.- E[SyncLeap arrives]
```
Comment (Fraser999) [inline node/src/components/sync_leaper.rs]: Did you mean for this to be a dotted line?

File: node/src/reactor/main_reactor/catch_up.rs [hunk 2]
```diff
@@ -28,6 +28,49 @@ pub(super) enum CatchUpInstruction {
     CommitUpgrade,
 }
 
+/// This diagram represents how the components involved in the sync process interact with each
```
Comment (Fraser999) [inline node/src/reactor/main_reactor/catch_up.rs]: Should this chart maybe be applied to the definition of the `MainReactor` in `main_reactor.rs:84`?

File: node/src/reactor/main_reactor/reactor_state.rs [hunk 3]
```diff
@@ -4,9 +4,58 @@ use schemars::JsonSchema;
...
+///     Upgrading --> CatchUp
+///     ShutdownForUpgrade --> End
+/// ```
+/// ```mermaid
+/// flowchart TD
+///     style Start fill:#66ccff,stroke:#333,stroke-width:4px
+///     style End fill:#66ccff,stroke:#333,stroke-width:4px
+///     style F fill:#ffcc66,stroke:#333,stroke-width:4px
+///     style G fill:#ffcc66,stroke:#333,stroke-width:4px
+///     title[CatchUP process]
```
Comment (Fraser999) [inline node/src/reactor/main_reactor/reactor_state.rs]:
```suggestion
///     title[CatchUp process]
```

File: node/src/reactor/main_reactor/reactor_state.rs [hunk 4]
```diff
@@ -4,9 +4,58 @@ use schemars::JsonSchema;
...
+///     A --> BlockIdentifier
+///     A --> SyncedBlockIdentifier
+///     A --> LocalTip
+///     BlockHash --> E[process identifier in<br/>block accumulator]
+///     BlockIdentifier --> E
+///     SyncedBlockIdentifier --> E
+///     LocalTip --> E
+///     CaughtUp --> H[handle upgrade<br/>if needed]
+///     H --> End
+///     E -->|more data needed<br/>from network| Leap
```
Comment (Fraser999) [inline node/src/reactor/main_reactor/reactor_state.rs]: I think we want to try and convey that we want the data to let us sync a block close to or at tip.  Something like:
"more data needed from network to let us sync near tip".  Not sure if that's better or not though!

File: node/src/reactor/main_reactor/catch_up.rs (hunk 2 above)
Comment (rafal-ch) [reply node/src/reactor/main_reactor/catch_up.rs]: I'm reluctant to attach it directly to `MainReactor` as it is responsible for many more things (diagnostics port, even stream, metrics...). The chart is focused on the interactions between components used in the so called "fast-sync" process.

However, I agree that the current location isn't perfect, so maybe I can move it to `MainReactor` and just add an appropriate comment about what is shown in the graph, wdyt?

File: node/src/components/sync_leaper.rs (hunk 1 above)
Comment (rafal-ch) [reply node/src/components/sync_leaper.rs]: That's a remnant from one of my previous approaches - I changed it to a normal arrow as it should be right now.

09ca60ebb01c4151d3d19c75ec6b9715ca663d81

File: node/src/reactor/main_reactor/reactor_state.rs (hunk 4 above)
Comment (rafal-ch) [reply node/src/reactor/main_reactor/reactor_state.rs]: I added more info as suggested. It's ok and we also have plenty of place in this flowchart :)

09ca60ebb01c4151d3d19c75ec6b9715ca663d81

File: node/src/reactor/main_reactor/reactor_state.rs (hunk 3 above)
Comment (rafal-ch) [reply node/src/reactor/main_reactor/reactor_state.rs]: Updated in https://github.com/casper-network/casper-node/commit/09ca60ebb01c4151d3d19c75ec6b9715ca663d81

File: node/src/reactor/main_reactor/catch_up.rs (hunk 2 above)
Comment (Fraser999) [reply node/src/reactor/main_reactor/catch_up.rs]: I like that suggestion.

**Review by Fraser999 — APPROVED** [review]
Optimistic approval given the suggested change is made.

File: node/src/reactor/main_reactor/catch_up.rs (hunk 2 above)
Comment (rafal-ch) [reply node/src/reactor/main_reactor/catch_up.rs]: Updated in 8ab8c97a34adda6010b29cb828ad51e9bb6832c6.

This is how it now looks like in the docs:

<kbd> </kbd>

**Pull request merged**